.github/scripts/rust/lint.rs

```diff
@@ -0,0 +1,281 @@
+    let prefix = contents.lines().take_while(|value| value.trim() != PREFIX);
+    let suffix = contents.lines().skip_while(|value| value.trim() != SUFFIX);
+
+    let indent = contents.lines()
+        .find(|value| value.trim() == PREFIX)
+        .map(|value| " ".repeat(value.len() - value.trim_start().len()))
+        .unwrap_or_else(|| "    ".to_owned());
+
+    let body = lints.forbid.into_iter().map(|lint| format!(r#"{indent}"-F{lint}","#))
+        .chain(lints.deny.into_iter().map(|lint| format!(r#"{indent}"-D{lint}","#)))
+        .chain(lints.force_warn.into_iter().map(|lint| format!(r#"{indent}"--force-warn{lint}","#)))
+        .chain(lints.warn.into_iter().map(|lint| format!(r#"{indent}"-W{lint}","#)))
+        .chain(lints.allow.into_iter().map(|lint| format!(r#"{indent}"-A{lint}","#)));
+
+    let contents: Vec<_> = prefix.map(|line| line.to_owned())
+        .chain(once(format!("{indent}## START CLIPPY LINTS ##")))
+        .chain(body)
+        .chain(suffix.map(|line| line.to_owned()))
+        .collect();
+
+    fs::write(&path, contents.join("\n")).expect("unable to write `.cargo/config.toml`");
+}
+
+fn print_diff(level: &str, left: &HashSet<String>, right: &HashSet<String>) {
+    let create = left - right;
+    let remove = right - left;
+
+    eprintln!("Lint Level {level} Errors:");
+    eprintln!("  Extra lints: {:?}", remove);
+    eprintln!("  Missing lints: {:?}", create);
+    eprintln!("  Run `cargo make update-clippy update` to automatically update the config.")
+}
+
+fn check(cwd: &Path) {
+    let lints = collect_lints(cwd);
+    let cargo = collect_cargo(cwd);
+
+    let mut fail = false;
+
+    if !(&lints.allow ^ &cargo.allow).is_empty() {
+        fail = true;
+        print_diff("Allow", &lints.allow, &cargo.allow);
+    }
+
+    if !(&lints.warn ^ &cargo.warn).is_empty() {
+        fail = true;
+        print_diff("Warn", &lints.warn, &cargo.warn);
+    }
+
+    if !(&lints.force_warn ^ &cargo.force_warn).is_empty() {
+        fail = true;
+        print_diff("Force Warn", &lints.force_warn, &cargo.force_warn);
+    }
+
+    if !(&lints.deny ^ &cargo.deny).is_empty() {
+        fail = true;
+        print_diff("Deny", &lints.deny, &cargo.deny);
+    }
+
+    if !(&lints.forbid ^ &cargo.forbid).is_empty() {
+        fail = true;
+        print_diff("Forbid", &lints.forbid, &cargo.forbid);
+    }
+
+    if fail {
+        std::process::exit(1);
+    } else {
+        println!("No changes necessary, we're good to go!");
+    }
+}
+
+fn main() {
+    let cwd = env::var("CARGO_MAKE_WORKSPACE_WORKING_DIRECTORY").expect("environment variable should exist");
+    let cwd = PathBuf::from(cwd);
+
+    let args = env::var("CARGO_MAKE_CLIPPY_LINT_MODE").or_else(|_| env::var("CARGO_MAKE_TASK_ARGS")).expect("environment variable should exist");
+    let mode = args.split(" ").next().expect("at least one position argument specifying mode required");
+
+    match mode {
+        "apply" => apply(&cwd),
```

Comment:
How do you feel about "generate"? "apply" sounds like you are fixing the lints

Reply:
fixed in [6b76071](https://github.com/hashintel/hash/pull/2075/commits/6b76071f8ced88156189b74f604d7d4e34588c67)